src/components/InfiniteGrid.js

```diff
@@ -0,0 +1,224 @@
+var React = require('react/addons');
+var PureRenderMixin = React.addons.PureRenderMixin;
+
+/**
+ * Inspired from: https://github.com/ggordan/react-infinite-grid
+ *
+ */
+
+var InfiniteGrid = React.createClass({
+
+  mixins: [ PureRenderMixin ],
+
+  propTypes: {
+    entries: React.PropTypes.arrayOf(React.PropTypes.element).isRequired,
+    maxEntries: React.PropTypes.number,
+    height: React.PropTypes.number,
+    width: React.PropTypes.number,
+    padding: React.PropTypes.number,
+    wrapperHeight: React.PropTypes.number,
+    lazyCallback: React.PropTypes.func,
+    renderRangeCallback: React.PropTypes.func,
+  },
+
+  getDefaultProps: function() {
+    return {
+      padding: 0,
+      entries: [],
+      maxEntries: 1,
+      height: 250,
+      width: 250,
+      bufferLength: 4
+    };
+  },
+
+  getInitialState: function() {
+    return {
+      initiatedLazyload: false,
+      minHeight: window.innerHeight * 2,
+      minItemIndex: 0,
+      maxItemIndex: 100,
+      itemDimensions: {
+        height: this._itemHeight(),
+        width: this._itemWidth(),
+        gridWidth: 0,
+        itemsPerRow: 2,
+      },
+    };
+  },
+
+  // METHODS
+
+  _wrapperStyle: function() {
```

Comment:
Could we move this to less?

Eventually I'm hoping to have all styles in Javascript, like this, but I think consistency is important.